cpp/modmesh/math/Complex.hpp

```diff
@@ -121,13 +135,23 @@ struct ComplexImpl
 template <typename T>
 bool operator<(const ComplexImpl<T> & lhs, const ComplexImpl<T> & rhs)
 {
-    return lhs.norm() < rhs.norm();
+    // Reference1: https://github.com/numpy/numpy/issues/12943
```

Comment:
It is good to have related reference like these, but they do not specify how numpy compares complex number (directly, not by using absolute number).

In the comment of the number issue 12943, https://github.com/numpy/numpy/issues/12943#issuecomment-969859179 , the use of lexicographic ordering of the real and imaginary parts was mentioned, it will be better to find a more explicit reference of the specification and also write down the code comments here.